### DISCUSSION TITLE
What is the correct way to setup VSCode for cross development of a windows program on linux

### DISCUSSION THREAD

**Discussion by richbai90** [microsoft/vscode-cpptools#9348] [Q&A]
## Background

I'm trying to write a simple program in C++ that will run on a windows machine. I run Linux as a daily and don't often develop for windows, so I am trying to cross compile using the mingw-w64 toolchain. The problem I'm running into is how to configure VSCode for the task. Here is my current setup.

## Problem

Given the setup below, when I try to include something like `<windows.h>` I get this error

> #include errors detected. Please update your includePath. Squiggles are disabled for this translation unit (/workspaces/filewatcher/src/include/FileWatcher.h).C/C++(1696)
cannot open source file "windows.h"C/C++(1696)

## Setup

* Debian 11 (bullsey) with build-essential and cmake installed
* mingw-w64 meta package installed
* VSCode cpptools extension installed and configured as shown

**cpp extension settings**
```js
/**
* c_cpp_properties.json
* Linux setup uses clang
* Windows setup uses mingw-w64
*/

{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**",
                "${workspaceFolder}/src/include"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/clang",
            "cStandard": "c17",
            "cppStandard": "c++14",
            "intelliSenseMode": "linux-clang-x64"
        },
        {
            "name": "win64",
            "includePath": [
                "${workspaceFolder}/**",
                "/usr/share/mingw-w64/**"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/x86_64-w64-mingw32-g++-win32",
            "cStandard": "c17",
            "cppStandard": "c++17",
            "intelliSenseMode": "windows-gcc-x64"
        }
    ],
    "version": 4
}
```
**Workspace setup**
```
.
├── dist
└── src
    ├── include
    │   └── winsdk // headers from the windows sdk that don't come with mingw 
    └── lib
```

## Other details

**VSCode info**
Version: 1.66.2
Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
Date: 2022-04-11T07:49:24.808Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Linux x64 5.17.4-1-default

**Extension list**

![image](https://user-images.githubusercontent.com/8270654/169706589-0ee859b7-c8ff-4845-9d52-5dbab886fc9c.png)

**Comment by sean-mcmanus** [comment]
This issue sounds similar to https://github.com/microsoft/vscode-cpptools/issues/9338 to you might try the solution in that issue, but in your case you're using a different compiler. I'm not familiar with using mingw to build on Linux for Windows, so I'm not sure if that even works or how it should be configured. You can run C/C++: Log Diagnostics to see what configuration is currently being used. If your compiler is supposed to compiling as cl.exe does then you'd want to make sure the intelliSenseMode is windows-msvc-x64 and set the compilerPath to "" so it doesn't try quering the compiler for system defines/includes and manually add the system includes. @Colengms might know more.
